### PR TITLE
✨clusterctl: upgrade to a specifc set of provider versions

### PR DESCRIPTION
**Pull request by fabriziopandini**
**What this PR does / why we need it**:
This PR add supports for upgrade to a specifc set of provider versions (vs upgrading to the latest version for all the providers) 

**Which issue(s) this PR fixes**:
Fixes https://github.com/kubernetes-sigs/cluster-api/issues/2501

/assign @vincepri @ncdc
/area clusterctl

### CODE REVIEW

File: cmd/clusterctl/pkg/client/cluster/upgrader.go [hunk 1]
```diff
@@ -197,6 +217,101 @@ func (u *providerUpgrader) getManagementGroup(coreProvider clusterctlv1.Provider
...
+	// Gets the API Version of Cluster API (contract).
+	// The this is required to ensure all the providers in a management group are consistent with the contract supported by the core provider.
+	// e.g if the core provider is v1alpha3, all the provider in the same management group should be v1alpha3 as well.
+
+	// The target contract is derived from the current version of the core provider, or, if the core provider is included in the upgrade list,
+	// from its target version.
+	targetCoreProviderVersion := managementGroup.CoreProvider.Version
+	for _, providerToUpgrade := range upgradeItems {
+		if providerToUpgrade.InstanceName() == managementGroup.CoreProvider.InstanceName() {
+			targetCoreProviderVersion = providerToUpgrade.NextVersion
```
Comment (ncdc) [inline cmd/clusterctl/pkg/client/cluster/upgrader.go]: You could `break` after this

**Comment by ncdc** [comment]
👀

File: cmd/clusterctl/pkg/client/cluster/upgrader.go (hunk 1 above)
Comment (fabriziopandini) [reply cmd/clusterctl/pkg/client/cluster/upgrader.go]: +1

**Pull request merged**